src/request.ts

```diff
@@ -27,3 +34,34 @@ export function fetch(token: string, config: AxiosRequestConfig) {
     ...config,
   });
 }
+
+export async function download(token: string, url: string, destPath: string, proxyUrl?: URL) {
```

Comment:
I created a `file.ts` in my PR that contains this plus the checksum logic. we can sync on how we want to organize it.

My perspective was to contain all the checksum in the same method when we do the download. If downloading lives in the request, we need to decouple the two steps which might make sense, does the old cli flow not use an md5 checksum?

Reply:
> does the old cli flow not use an md5 checksum?

nope it does not, so I do think we need to decouple the two